great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1211,6 +1228,14 @@ def get_batch_data_and_markers(
         RuntimeQueryBatchSpec (illegal type "{str(type(batch_spec))}" was received).
                         """
             )
+        has_query = 1 if batch_spec.get("query") else 0
```

Comment:
@billdirks While equivalent, I think that the following style, used in other parts of the codebase, is more elegant in the sense that it always uses booleans to compute the mutual exclusion as a boolean (i.e., without the integers):

```
        if (
            sum(
                bool(x)
                for x in [
                    batch is not None,
                    batch_list is not None,
                    batch_request is not None,
                    batch_request_list is not None,
                ]
            )
            > 1 
        ):
            raise ValueError(
                "No more than one of batch, batch_list, batch_request, or batch_request_list can be specified"
            )
```

Your call.  Thanks for considering.